modules/tensor_mechanics/include/auxkernels/RankTwoScalarAux.h

```diff
@@ -29,12 +29,18 @@ class RankTwoScalarAux : public AuxKernel
 
 protected:
   virtual Real computeValue();
-  Real calcEigenValues();
 
   const MaterialProperty<RankTwoTensor> & _tensor;
   MooseEnum _scalar_type;
 
+  /// whether or not selected_qp has been set
+  const bool _has_selected_qp;
+
+  /// The std::vector will be evaluated at this quadpoint only
+  const unsigned int _selected_qp;
+
 private:
```

Comment:
(i've made it protected).  A general question: i always try to make things private, to keep them inside the class and protect them from anyone else.  Is that typically not a good idea?